core/src/main/java/io/aiven/kafka/tieredstorage/metrics/CaffeineStatsCounter.java

```diff
@@ -39,184 +40,180 @@ public class CaffeineStatsCounter implements StatsCounter {
     }
 
     @Override
     public void recordMisses(@NonNegative final int count) {
-        cacheMissesSensor.record(count);
+        cacheMissCount.add(count);
     }
 
     @Override
     public void recordLoadSuccess(@NonNegative final long loadTime) {
-        cacheLoadSuccessSensor.record();
-        cacheLoadSuccessTimeSensor.record(loadTime);
+        cacheLoadSuccessCount.increment();
+        cacheLoadSuccessTimeTotal.add(loadTime);
     }
 
     @Override
     public void recordLoadFailure(@NonNegative final long loadTime) {
-        cacheLoadFailureSensor.record();
-        cacheLoadFailureTimeSensor.record(loadTime);
+        cacheLoadFailureCount.increment();
+        cacheLoadFailureTimeTotal.add(loadTime);
     }
 
     @Override
     public void recordEviction(@NonNegative final int weight, final RemovalCause cause) {
-        cacheEvictionSensor.record();
-        evictionSensorByCause(cause).record();
-        cacheEvictionWeightSensor.record(weight);
-        evictionWeightSensorByCause(cause).record(weight);
-    }
-
-    Sensor evictionSensorByCause(final RemovalCause cause) {
-        return new SensorProvider(metrics, "cause." + cause.name() + "." + CACHE_EVICTION,
-            () -> Map.of("cause", cause.name()))
-            .with(metricCacheEvictionRateByCause, new Rate())
-            .with(metricCacheEvictionTotalByCause, new CumulativeSum())
-            .get();
-    }
-
-    Sensor evictionWeightSensorByCause(final RemovalCause cause) {
-        return new SensorProvider(metrics, "cause." + cause.name() + "." + CACHE_EVICTION_WEIGHT,
-            () -> Map.of("cause", cause.name()))
-            .with(metricCacheEvictionWeightByCause, new Rate())
-            .with(metricCacheEvictionWeightTotalByCause, new CumulativeSum())
-            .get();
+        cacheEvictionCountTotal.increment();
+        cacheEvictionCountByCause.get(cause).increment();
+        cacheEvictionWeightTotal.add(weight);
+        cacheEvictionWeightByCause.get(cause).add(weight);
     }
 
     @Override
     public CacheStats snapshot() {
         return CacheStats.of(
-            (long) metrics.metric(metrics.metricInstance(metricCacheHitsTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheMissesTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheLoadSuccessTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheLoadFailureTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheLoadSuccessTimeTotal)).metricValue()
-                + (long) metrics.metric(metrics.metricInstance(metricCacheLoadFailureTimeTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheEvictionTotal)).metricValue(),
-            (long) metrics.metric(metrics.metricInstance(metricCacheEvictionWeightTotal)).metricValue()
+            negativeToMaxValue(cacheHitCount.sum()),
```

Comment:
AFAIU, `negativeToMaxValue` is some quirk of the Caffeine implementation, but maybe leave a comment why, if known?

Reply:
The idea was to return `Long.MAX_VALUE` in case of overflow since `CaffeineStats` does not accept negative values. We are not validating them again, we are bypassing the `validation` done by Caffeine since we are not using `ConcurrentStatsCounter` but reimplementing it. I'll add a comment.

Reply:
Fixed.